tests/testutils.go

```diff
@@ -0,0 +1,202 @@
+package tests
+
+const KrmFunctionSchema = `swagger: "2.0"
+info:
+  title: KRM Function Metadata
+  version: v1alpha1
+definitions:
+  KRMFunctionDefinition:
+    type: object
+    description: |
+      KRMFunctionDefinition is metadata that defines a KRM function
+      the same way a CustomResourceDefinition defines a custom resource.
+    x-kubernetes-group-version-kind:
+      - group: config.kubernetes.io
+        kind: KRMFunctionDefinition
+        version: v1alpha1
+    required:
+      - apiVersion
+      - kind
+      - spec
+    properties:
+      apiVersion:
+        description: apiVersion of KRMFunctionDefinition. i.e. config.kubernetes.io/v1alpha1
+        type: string
+        enum:
+          - config.kubernetes.io/v1alpha1
+      kind:
+        description: kind of the KRMFunctionDefinition. It must be KRMFunctionDefinition.
+        type: string
+        enum:
+          - KRMFunctionDefinition
+      spec:
+        type: object
+        description: spec contains the metadata for a KRM function.
+        required:
+          - group
+          - names
+          - description
+          - publisher
+          - versions
+        properties:
+          group:
+            description: group of the functionConfig
+            type: string
+          description:
+            description: brief description of the KRM function.
+            type: string
+          publisher:
+            description: the entity (e.g. organization) that produced and owns this KRM function.
+            type: string
+          names:
+            description: the resource and kind names for the KRM function
+            type: object
+            required:
+            - kind
+            properties:
+              kind:
+                description: the Kind of the functionConfig
+                type: string
+          versions:
+            description: the versions of the functionConfig
+            type: array
+            items:
+              type: object
+              required:
+                - name
+                - schema
+                - idempotent
+                - runtime
+                - usage
+                - examples
+                - license
+              properties:
+                name:
+                  description: Version of the functionConfig
+                  type: string
+                schema:
+                  description: a URI pointing to the schema of the functionConfig
+                  type: object
+                  required:
+                    - openAPIV3Schema
+                  properties:
+                    openAPIV3Schema:
+                      description: openAPIV3Schema is the OpenAPI v3 schema to use for validation
+                      # kube-openapi validation doesn't support references, and inlining this ref is extremely tedious
+                      # $ref: "#/definitions/io.k8s.apiextensions-apiserver.pkg.apis.apiextensions.v1.JSONSchemaProps"
```

Comment:
It appears that [kube-openapi/validate.AgainstSchema](https://pkg.go.dev/k8s.io/kube-openapi@v0.0.0-20220124234850-424119656bbf/pkg/validation/validate#AgainstSchema) doesn't support references and inlining this reference here will be a lot of tedious work, so I commented it out here.

If it is important to validate this field, it would be helpful to have some guidance for how to do the validation, or I can also do the tedious work of inlining the reference if necessary.

@mengqiy